File: 404-sum-of-left-leaves/404. Sum of Left Leaves.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int sumOfLeftLeaves(TreeNode* root) {
        if(root==NULL){
            return 0;

        }
        queue<TreeNode*>q;
        q.push(root);
        int sum =0;
        while(!q.empty()){
            int size = q.size();
            for(int i=0;i<size;i++){
                TreeNode * front = q.front();
            q.pop();
            if(front->left){
                if(front->left->left ==nullptr && front->left->right ==nullptr)
                sum+=front->left->val;
                q.push(front->left);
            }

            if(front->right){
                q.push(front->right);
            }
            }
        }
        return sum;
    }
};

File: 404-sum-of-left-leaves/404. Sum of Left Leaves.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    int sum = 0;
    public void solve1(TreeNode root){
        if(root== null) {
            return;
        }
        
        if(root.left!=null){
            if(root.left.left == null && root.left.right == null) {
            sum += root.left.val;
        }
        }
        solve1(root.left);
        solve1(root.right);

    }
    public int sumOfLeftLeaves(TreeNode root) {
        solve1(root);
        return sum;

    }
}